src/celeritas/io/ImportDataTrimmer.cc

```diff
@@ -0,0 +1,376 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/io/ImportDataTrimmer.cc
+//---------------------------------------------------------------------------//
+#include "ImportDataTrimmer.hh"
+
+#include <algorithm>
+#include <utility>
+
+#include "corecel/Assert.hh"
+#include "corecel/cont/Range.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+struct ImportDataTrimmer::GridFilterer
+{
+    size_type stride;
+    size_type orig_size;
+
+    // Whether to keep the data at this index
+    inline bool operator()(size_type i) const;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from input parameters.
+ */
+ImportDataTrimmer::ImportDataTrimmer(Input const& inp) : options_{inp}
+{
+    CELER_EXPECT(options_.max_size >= 2);
+}
+
+//---------------------------------------------------------------------------//
+void ImportDataTrimmer::operator()(ImportData& data)
+{
+    if (options_.materials)
+    {
+        // Reduce the number of materials, elements, etc.
+        (*this)(data.isotopes);
+        (*this)(data.elements);
+        (*this)(data.geo_materials);
+        (*this)(data.phys_materials);
+
+        (*this)(data.regions);
+        (*this)(data.volumes);
+
+        (*this)(data.sb_data);
+        (*this)(data.livermore_pe_data);
+        (*this)(data.neutron_elastic_data);
+        (*this)(data.atomic_relaxation_data);
+
+        (*this)(data.optical_materials);
+    }
+
+    if (options_.physics)
+    {
+        // Reduce the number of physics processes
+        (*this)(data.particles);
+        (*this)(data.processes);
+        (*this)(data.msc_models);
+    }
+
+    if (options_.mupp)
+    {
+        // Reduce the resolution of the muon pair production table
+        (*this)(data.mu_pair_production_data);
+    }
+
+    this->for_each(data.elements);
+    this->for_each(data.geo_materials);
+    this->for_each(data.phys_materials);
+
+    this->for_each(data.processes);
+    this->for_each(data.msc_models);
```

Comment:
@sethrj these will cause the `ProcessBuilder` test to fail and change the results in the MSC tests. What do you want to do about those?

Reply:
Uh oh, I clearly didn't think that through. Maybe all the `for_each` should live inside the `if (options_.physics)`? Or we need an option to trim "number of processes" vs "contents of each process"? I think the former should let you trim the mupp table without breaking anything else. Could you try that on your branch? Sorry about that.